app/models/service_order.rb

```diff
@@ -0,0 +1,15 @@
+class ServiceOrder < ActiveRecord::Base
+  belongs_to :tenant
+  belongs_to :user
+  has_many :miq_requests, :dependent => :nullify, :after_add => :assign_user_name_to
+
+  validates :state, :inclusion => {:in => %w(wish cart ordered)}
+  validates :name, :presence => true
+  validates :state, :presence => true
+
+  default_value_for :user_id, User.current_user.id
+
+  def assign_user_name_to(miq_request)
+    miq_request.user_name = user.name
```

Comment:
We have not worked out the details of how a wish-list work would just yet so I would not code for that now.  I would say when we create the ServiceOrder we will be passing the user object to the create call and we can use that object to capture the `user_name` as well as the `tenant_id` at that time.  Then the caller will be responsible for passing the user object with the proper identify.